src/parsing.jl

```diff
@@ -235,18 +235,46 @@ function Base.parse(::Type{ZonedDateTime}, pqv::PQValue{PQ_SYSTEM_TYPES[:timesta
     end
 
     for fmt in TIMESTAMPTZ_FORMATS[1:end-1]
-        try
-            return parse(ZonedDateTime, str, fmt)
-        catch
-            continue
-        end
+        parsed = tryparse(ZonedDateTime, str, fmt)
+        parsed !== nothing && return parsed
     end
     # Cut off digits after the third after the decimal point,
     # since DateTime in Julia currently handles only milliseconds, see Issue #33
     str = replace(str, r"(\.[\d]{3})\d+" => s"\g<1>")
     return parse(ZonedDateTime, str, TIMESTAMPTZ_FORMATS[end])
 end
 
+_DEFAULT_TYPE_MAP[:date] = Date
+function Base.parse(::Type{Date}, pqv::PQValue{PQ_SYSTEM_TYPES[:date]})
+    str = string_view(pqv)
+
+    if str == "infinity"
+        return typemax(Date)
+    elseif str == "-infinity"
+        return typemin(Date)
+    end
+
+    return parse(Date, str)
+end
+
+_DEFAULT_TYPE_MAP[:time] = Time
+function Base.parse(::Type{Time}, pqv::PQValue{PQ_SYSTEM_TYPES[:time]})
+    str = string_view(pqv)
+
+    try
+        return parse(Time, str)
+    catch err
+        if !(err isa InexactError)
+            rethrow(err)
+        end
+    end
+
+    # Cut off digits after the third after the decimal point,
+    # since Time in Julia currently handles only milliseconds, see Issue #33
+    str = replace(str, r"(\.[\d]{3})\d+" => s"\g<1>")
```

Comment:
This issue is long closed I think?
What #33 on what repo?

Reply:
#33 on this repo, which should be the assumption when no repo is provided

That issue resulted in this replacement, so I linked that as a reference

Reply:
My assumptiom was that it was a reference to a bug in Julia or in TimeZones.
Since it doesn't seem to be related to this.

Reply:
Is it worth moving this regex and comment into a function?